src/yieldplotlib/core/single_inputs.py

```diff
@@ -0,0 +1,38 @@
+"""The base `SingleInput` class.
+
+`SingleInput` is a class designed to facilitate the reading, setting, and manipulation of
+various types of yield inputs. It inherits from dict for direct access to
+built-in dictionary functions such as get(), update(), etc.
+"""
+
+
+class SingleInput(dict):
+    """Base class for all Inputs."""
+
+    def __init__(self, *args, **kwargs):
+        super(SingleInput, self).__init__(*args, **kwargs)
+
+    def apply(self, key, func):
+        """Apply a function to all values of a given key."""
+        try:
+            self.update(key=func(self.get(key)))
```

Comment:
I've never really used the `update` function, is there a significant benefit of using it instead of `self[key] = func(self.get[key])`? A quick google showed there's a performance benefit to using the key/value pair for some reason: https://stackoverflow.com/questions/15456158/python-dict-update-vs-subscript-to-add-a-single-key-value-pair

Reply:
I think the main benefit is that you can set multiple key/value pairs at a time with update (without needing an external loop). You're right though that as written it is just updating a single key so I should probably just set that directly for the performance reasons you linked